### PR TITLE
Move Multi-user under Cluster section of the wizard

### PR DESCRIPTION
**Pull request by tmscarla**
## Description

In order to enhance PCUI user experience, we need to thin out the steps of our cluster creation wizard. Following UX design, the whole Multi-user step needs to be removed, and Multi-user section needs to be placed under `Cluster` step of the wizard.

This PR aims at solving the issue, by conditionally showing Multi-user properties in Cluster step through a dedicated checkbox.

## Changelog entry

* Remove `Multi-user` wizard step and embedded it into `Cluster` step in a dedicated section

## How Has This Been Tested?
* Unit tests
* E2E tests
* Manually navigated through the end of the wizard
   * With multi-user enabled
   * With multi-user disabled

## References
* https://cloudscape.design/components/checkbox/

## PR Quality Checklist

- [x] I added tests to new or existing code
- [x] I removed hardcoded strings and used our `i18n` solution instead (see [here](https://github.com/aws-samples/pcluster-manager/pull/175/commits/fdc6b77987c87a26f51dbc8da5d371d95ef80601))
- [ ] I made sure no sensitive info gets logged at any time in the codebase (see [here](https://cheatsheetseries.owasp.org/cheatsheets/Logging_Cheat_Sheet.html)) (e.g. no user info or details, no stacktraces, etc.)
- [ ] I checked that infrastructure/update_infrastructure.sh runs without any error
- [x] I checked that `npm run build` builds without any error
- [x] I checked that clusters are listed correctly
- [x] I checked that a new cluster can be created (config is produced and dry run passes)
- [x] I checked that login and logout work as expected

In order to increase the likelihood of your contribution being accepted, please make sure you have read both the [Contributing Guidelines](../CONTRIBUTING.md) and the [Project Guidelines](../PROJECT_GUIDELINES.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Screenshot 2023-01-09 at 17 45 40

### CODE REVIEW

File: frontend/src/old-pages/Configure/__tests__/useWizardNavigation.test.tsx [hunk 1]
```diff
@@ -157,122 +157,6 @@ describe('Given a hook to navigate through the wizard', () => {
     })
   })
 
-  describe('when on cluster page', () => {
```
Comment (BarcoMasile) [inline frontend/src/old-pages/Configure/__tests__/useWizardNavigation.test.tsx:160]: question: the tests deleted from here, shouldn't they be still present somewhere else?

File: frontend/src/old-pages/Configure/Cluster.tsx [hunk 2]
```diff
@@ -404,40 +415,44 @@ function Cluster() {
...
-    </Container>
+    <SpaceBetween direction="vertical" size="s">
+      <Container>
+        <SpaceBetween direction="vertical" size="s">
+          <RegionSelect />
+          <OsSelect />
+          <VpcSelect />
+          <CustomAMISettings
+            basePath={configPath}
+            appPath={['app', 'wizard']}
```
Comment (BarcoMasile) [inline frontend/src/old-pages/Configure/Cluster.tsx:436]: suggestion(non-blocking): this array `['app', 'wizard']` can easily be put outside the re-rendered template. This way you would pass the same reference avoiding useless re-renders

**Review by mtfranchetto — APPROVED** [review]

File: frontend/src/old-pages/Configure/__tests__/useWizardNavigation.test.tsx (hunk 1 above)
Comment (tmscarla) [reply frontend/src/old-pages/Configure/__tests__/useWizardNavigation.test.tsx:160]: It covered the `multiUser` case: if enabled, you navigate to multiUser page, headNode otherwise.

Moving multiUser to cluster page eliminates this edge case.
Standard flow has already been tested in other test cases.
